implement cloud bench module

nkraemer-sysdig opened the pull request:
Add cloudbench module

nkraemer-sysdig commented on examples/organizational_cloudvision/versions.tf:
```diff
@@ -4,5 +4,9 @@ terraform {
     }
+    sysdig = {
+      source = "sysdiglabs/sysdig"
+      version = "0.5.17" // TODO release updated sysdig provider
```
Once https://github.com/sysdiglabs/terraform-provider-sysdig/pull/111 is merged, we will need to release this new version of the sysdig provider.

nkraemer-sysdig commented on main.tf:
```diff
@@ -31,4 +36,6 @@ module "services" {
   services_assume_role_arn = aws_iam_role.cloudvision_role.arn
   tags                     = var.tags
+
+  cloudbench_account_ids = [var.aws_organization_cloudvision_account_id] // TODO populate this with all accounts in org
```
Need to sync on how to determine list of accountIDs. Is it possible to fetch the list of accounts within an orginization? Will we have permissions to create roles/policies in these child accounts?

nkraemer-sysdig replied:
Perfect! I'll use the same approach to create the cloud-bench role in the same member account, and worry about multiple accounts in a separate PR

nkraemer-sysdig commented on modules/services_cloud_bench/main.tf:
```diff
@@ -0,0 +1,90 @@
+    actions = ["sts:AssumeRole"]
+    principals {
+      type = "AWS"
+      identifiers = ["arn:aws:iam::059797578166:user/noah.kraemer"] // TODO fix this
```
This needs to be updated once we provision an internal user that cloud-bench will use.

nkraemer-sysdig replied:
Yep, that's a great idea! I think it makes sense to use a datasource from the `sysdig_secure_cloud_account` provider, since I don't think we want to allow anything except the few known sysdig accounts. And yes, these will likely be different for different SAAS regions but will be the same for all customers

nkraemer-sysdig commented on modules/services_cloud_bench/main.tf:
```diff
@@ -0,0 +1,90 @@
+data "aws_iam_policy_document" "permissions" {
+  statement {
+    effect = "Allow"
+    actions = [
```
Actual list of permissions needs review/subject to change

wideawakening reviewed (commented):
forgot to push submit review :neutral_face:

nkraemer-sysdig commented on modules/services/main.tf:
```diff
@@ -55,3 +55,11 @@ module "cloud_connector" {
+module "cloud_bench" {
+  source = "../services_cloud_bench"
+
+  for_each   = toset(var.cloudbench_account_ids)
```
Okay, going to remove it for now and deal with multi-account in a separate PR

nkraemer-sysdig commented:
Thanks for the comments @wideawakening! I opened https://github.com/sysdiglabs/terraform-aws-cloudvision/pull/8 to replace this to avoid the rebase, and I have addressed your comments there.
